musicbot/songinfo.py

```diff
@@ -31,9 +29,9 @@ def format_output(self, playtype):
             embed.add_field(name=config.SONGINFO_UPLOADER,
                             value=self.uploader, inline=False)
 
-            if self.duration is not None:
+            if self.duration is not None and type(self.duration) is not float:
```

Comment:
Checking for a float causes all SoundCloud songs return unknown for duration.

I propose that the int conversion happen on line 17 when self.duration is defined
`self.duration = int(duration)`
```suggestion
            if self.duration is not None:
```